arm9/src/list_menu.c

```diff
@@ -0,0 +1,572 @@
+#include "global.h"
+#include "heap.h"
+#include "main.h"
+#include "list_menu.h"
+#include "text.h"
+
+void ListMenuPrintEntries(struct ListMenu * list, u16 startIndex, u16 yOffset, u16 count);
+void ListMenuDrawCursor(struct ListMenu * list);
+BOOL ListMenuChangeSelection(struct ListMenu * list, u8 updateCursorAndCallCallback, u8 count, u8 movingDown);
+void ListMenuCallSelectionChangedCallback(struct ListMenu * list, u8 onInit);
+
+static inline u32 MakeFontColor(u32 fgPal, u32 shdwPal, u32 bgPal)
+{
+    return (u32)(
+        ((u32)(fgPal << 24) >> 8)
+        | ((u32)(shdwPal << 24) >> 16)
+        | ((u32)(bgPal << 24) >> 24)
+    );
+}
+
+THUMB_FUNC struct ListMenu * ListMenuInit(const struct ListMenuTemplate * template, u16 cursorPos, u16 itemsAbove, u32 heap_id)
+{
+    struct ListMenu * list = AllocFromHeap(heap_id, sizeof(struct ListMenu));
+    list->template = *template;
+    list->cursor = ListMenuCursorNew(heap_id);
+    list->cursorPos = cursorPos;
+    list->itemsAbove = itemsAbove;
+    list->unk_30 = 0;
+    list->unk_31 = 0;
+    list->taskId = 0xFF;
+    list->unk_33 = 0;
+    list->heap_id = (u8)heap_id;
+    list->cursorPal = list->template.cursorPal;
+    list->fillValue = list->template.fillValue;
+    list->cursorShadowPal = list->template.cursorShadowPal;
+    list->lettersSpacing = list->template.lettersSpacing;
+    list->fontId = list->template.fontId;
+    list->overrideEnabled = FALSE;
+    if (list->template.totalItems < list->template.maxShowed)
+        list->template.maxShowed = list->template.totalItems;
+    ListMenuCursorSetColor(
+        list->cursor,
+//        MakeFontColor(list->template.cursorPal, list->template.cursorShadowPal, list->fillValue)
```

Comment:
what happened here?

Reply:
Using that static inline there does not match yet. But when it does, the commented line will replace the ones below.